repose-aggregator/src/docs/asciidoc/services/http-connection-pool.adoc

```diff
@@ -1,6 +1,183 @@
+<17> The truststore filename either fully qualified or relative to the Repose configuration directory (/etc/repose is the default).
+     Used when it is necessary to have the certificate of the requested resource's server. +
+     NOTE: The `truststore-filename` and `truststore-password` are mutually required. If one exists, then the other must exist.
+<18> The password to access the truststore. +
+     NOTE: The `truststore-filename` and `truststore-password` are mutually required. If one exists, then the other must exist.
+<19> Indicates that the configured list of headers should be added to every outbound request on this connection pool.
+     Use is optional.
+<20> A header to be added to every outbound request from this connection pool.
+     At least on is required when using the optional `headers` element.
+<21> A second pool is defined here that uses all the defaults.
+
+== Common Use Cases/Problems
+
+=== Origin Service doesn't support chunked encoding (e.g. WSGI)
+
+If the origin service doesn't support chunked encoding, it can be turned off easily.
+Add or update the `chunked-encoding` attribute for your default pool to false.
+
+[source,xml]
+.partial http-connection-pool.cfg
+----
+<pool id="default"
+      default="true" <!--1-->
+      ... <!--2-->
+      chunked-encoding="false"/> <!--3-->
+</pool>
+----
+<1> Be sure it's the default pool being updated.
+<2> Here's the rest of the pool attributes.
+<3> Add `chunked-encoding` if not already present and set it to false.
+
+[NOTE]
+This can be done for any other pool as well, if it's known that it will be used to communicate to a service that doesn't support chunked encoding.
+
+=== Need higher concurrency
+
+If the connection pool is filling up, but more resources are available on the machine *Repose* is running on, the pool can be expanded to allow more concurrent connections.
+This is most useful if wait times are high or requests are timing out.
+The size of the pools can be increased by raising `http.conn-manager.max-total` and `http.conn-manager.max-per-route`.
+
+[source,xml]
+.partial http-connection-pool.cfg
+----
+<pool id="default"
+      ... <!--1-->
+      http.conn-manager.max-total="800" <!--2-->
+      http.conn-manager.max-per-route="300"/> <!--3-->
+</pool>
+----
+<1> The existing pool attributes.
+<2> The new overall pool size.
+    This is the maximum number of connections this pool will have open at once.
+<3> The new max number of connections to a single endpoint.
+    If a pool is only ever used for a single endpoint (e.g. foo.com), then this number and `http.conn-manager.max-total` should be the same.
+    If they aren't then the max pool size will never be reached.
+
+=== Requested endpoint requires a static header
+
+Some services require a static header be present just for the purposes of identification, this can easily be done by adding the `headers` element to the pool configuration.
```

Comment:
You could also use a period, or a conjunction. At any rate, these are totally independent clauses.